### ISSUE TITLE
Ifdef with JS code trigger compile errors

### ISSUE THREAD

**Issue by edubart** [vnmakarov/mir#190]
C2M is failing to compile a header that contains __EMSCRIPTEN__ ifdefs, the Emscripten compiler has the macro `EM_JS` to emit JS code that uses string literals with single quotes, seems like C2M is unable to parse in such situations:

```c
#ifdef __EMSCRIPTEN__
EM_JS(void, listener, (void), {
  window.addEventListener('beforeunload', Module.beforeunload);
});
#endif

int main() {
  return 0;
}
```

The above, fails with the errors:
```
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
c/sokol-nuklear-calculator.c:3:27: multibyte character
```

**Comment by edubart** [comment]
I've noticed other piece of C code, that have this error, the following one (from the Lua language sources):
```c
#if 0
** you can update the following list with this command:
**
**  sed -n '/^OP_/\!d; s/OP_/\&\&L_OP_/ ; s/,.*/,/ ; s/\/.*// ; p'  lopcodes.h
**
#endif
```

**Comment by vnmakarov** [comment]
Here is the fix for this issue:

https://github.com/vnmakarov/mir/commit/f6c695f1d95f393cad61b967bd61ebf7b80d7855
and
https://github.com/vnmakarov/mir/commit/7caf638b2c3dec7200822d4ab4a68a9d602f145b

**Comment by rofl0r** [comment]
wasn't this the fix for this issue ? https://github.com/vnmakarov/mir/commit/3e9a44cc511a92a95344823bbe29e4618d183259
looks more likely

**Comment by vnmakarov** [comment]
> wasn't this the fix for this issue ? [3e9a44c](https://github.com/vnmakarov/mir/commit/3e9a44cc511a92a95344823bbe29e4618d183259)
> looks more likely

Yes, you are right.  The 1st commit I posted here was for issue #189.

**Comment by edubart** [comment]
It worked, thanks!
